stix_shifter_modules/azure_log_analytics/stix_transmission/connector.py

```diff
@@ -50,39 +83,254 @@ async def create_results_connection(self, query, offset, length):
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except HttpResponseError as ex:
+            if isinstance(ex.error, ODataV4Format):
+                if 'PathNotFoundError' in str(ex.error) or 'WorkspaceNotFoundError' in str(ex.error):
+                    response_dict['code'] = 404
+                    response_dict['message'] = "Invalid Parameter: workspaceId"
+                else:
+                    response_dict['code'] = 400
+                    response_dict['message'] = "Invalid Query"
+            else:
+                response_dict['code'] = 500
+                response_dict['message'] = "Invalid Host/Port"
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except ValueError as ex:
+            if "Invalid tenant" in str(ex):
+                response_dict['message'] = "Invalid Parameter: tenant"
+            else:
+                response_dict['message'] = "Invalid Parameter"
+            response_dict['code'] = 404
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except TimeoutError as ex:
+            response_dict['code'] = 408
+            response_dict['message'] = 'TimeoutError ' + str(ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
+
+        except Exception as ex:
+            response_dict['message'] = ex
+            self.logger.error('error when getting search results: %s', ex)
+            ErrorResponder.fill_error(return_obj, response_dict, ['message'], connector=self.connector)
         return return_obj
+
+    def format_response(self, data):
+        """ Format the API responses
+        :param data: list
+        :return: list """
+        self.severity = {value: key for key, value in self.config_map["enum_supported_values"]["Severity"].items()}
+
+        for row in data:
+            copied_row = copy.deepcopy(row)
+            try:
+                # formatting security alert response
+                if row.get("Type") == "SecurityAlert":
+                    row = self.format_security_alert(row)
+                # formatting security incident response
+                elif row.get("Type") == 'SecurityIncident':
+                    row = self.format_security_incident(row)
+                # formatting security event response
+                elif row.get("Type") == "SecurityEvent":
+                    if row.get('FileHash'):
+                        row = self.find_filehash_map(row)
+            except:
+                row = copied_row
+        return data
+
+    def format_security_alert(self, security_alert):
+        """ Format the security alert response
+        :param security_alert: dict
+        :return: dict """
+
+        if security_alert.get("Techniques") == "":
+            security_alert["Techniques"] = None
```

Comment:
> why do you need to replace the empty string?

We have observed that the field value comes empty in API output. To avoid empty value in stix output we are replacing it with 'None' so that it is skipped in result translation

Reply:
this is a bug in json to stix translator #1510 
you can remove this check. once this bug is fixed it should be automatically handled by the utility.

Reply:
> this is a bug in json to stix translator #1510 you can remove this check. once this bug is fixed it should be automatically handled by the utility.

Removed as per your suggestions